src/test/java/net/sf/jsqlparser/test/alter/AlterTest.java

```diff
@@ -252,4 +253,28 @@ public void testDropColumnRestrictIssue551() throws JSQLParserException {
         assertStatementCanBeDeparsedAs(stmt, "ALTER TABLE table1 DROP COLUMN NewColumn");
         
     }
+    
+    public void testAddConstraintKeyIssue320() throws JSQLParserException {
+        String tableName = "table1";
+        String columnName1 = "col1";
+        String columnName2 = "col2";
+        String columnName3 = "col3";
+        String columnName4 = "col4";
+        String constraintName1 = "table1_constraint_1";
+        String constraintName2 = "table1_constraint_2";
+
+        for(String constraintType : Arrays.asList("UNIQUE KEY", "KEY")) {
+            Statement stmt = CCJSqlParserUtil.parse("ALTER TABLE " + tableName + " ADD CONSTRAINT " + constraintName1 + " " + constraintType + " (" + columnName1 + ")");
```

Comment:
I merged and changed now this lines. You used assertStatementCanBeDeparsedAs which needs a separate parse step. Using assertSqlCanBeParsedAndDeparsed this is not needed anymore.